supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala

```diff
@@ -0,0 +1,225 @@
+        tesPaths.toContainerPath(f).valueString,
+        "File",
+        Some(false)
+      )
+    } ++ Seq(commandScript)
+
+  // TODO add TES logs to standard outputs
+  private val standardOutputs = Seq("rc", "stdout", "stderr").map {
+    f =>
+      TaskParameter(
+        f,
+        Some(fullyQualifiedTaskName + "." + f),
+        tesPaths.storageOutput(f),
+        tesPaths.containerOutput(f),
+        "File",
+        Some(false)
+      )
+  }
+
+  // extract output file variable names
+  private val outputFileNames = jobDescriptor.call.task.outputs
+      .filter {
+        case o => o.wdlType.toWdlString == "Array[File]" || o.wdlType.toWdlString == "File"
+      }
+    .map{case o => o.unqualifiedName}
+
+  // extract output files
+  private val outputWdlFiles = jobDescriptor.call.task
+    .findOutputFiles(jobDescriptor.fullyQualifiedInputs, NoFunctions)
+
+  private val wdlOutputs = outputFileNames.zip(outputWdlFiles)
+    .flatMap {
+      case (outputName, f: WdlSingleFile) => {
+        val outputFile = f.valueString
+        Seq(
+          TaskParameter(
+            outputName,
+            Some(fullyQualifiedTaskName + "." + outputName),
+            tesPaths.storageOutput(outputFile),
+            tesPaths.containerOutput(outputFile),
+            "File",
+            Some(false)
+          )
+        )
+      }
+      case (outputName, g: WdlGlobFile) => {
+        val globName = callEngineFunction.globName(g.value)
+        val globDirName = outputName + ".globDir"
+        val globDirectory = globName + "/"
+        val globListName = outputName + ".globList"
+        val globListFile = globName + ".list"
+        Seq(
+          TaskParameter(
+            globDirName,
+            Some(fullyQualifiedTaskName + "." + globDirName),
+            tesPaths.storageOutput(globDirectory),
+            tesPaths.containerOutput(globDirectory),
+            "Directory",
+            Some(false)
+          ),
+          TaskParameter(
+            globListName,
+            Some(fullyQualifiedTaskName + "." + globListName),
+            tesPaths.storageOutput(globListFile),
+            tesPaths.containerOutput(globListFile),
+            "File",
+            Some(false)
+          )
+        )
+      }
+    }
+
+  val outputs = wdlOutputs ++ standardOutputs
+
+  private val workingDirVolume = runtimeAttributes
+    .dockerWorkingDir
+    .map(path => Volume(
+      path,
+      // TODO all volumes currently get the same requirements
+      Some(runtimeAttributes.disk.to(MemoryUnit.GB).amount.toInt),
+      None,
+      path
+    ))
+
+  val volumes = Seq(
+    Volume(
+      tesPaths.dockerWorkflowRoot.toString,
+      Some(runtimeAttributes.disk.to(MemoryUnit.GB).amount.toInt),
+      None,
+      tesPaths.dockerWorkflowRoot.toString
+    )
+  ) ++ workingDirVolume
+
+  val resources = Resources(
+    runtimeAttributes.cpu,
+    runtimeAttributes.memory.to(MemoryUnit.GB).amount.toInt,
+    Some(false),
+    Some(volumes),
```

Comment:
This is something we picked up from my previous team where `null`s weren't verboten and we'd get bit by this now and then.`Option(null)` is `None` whereas `Some(null)` is exactly what it says. 

Generally my preference is to avoid things which the developer "knows" to be true in favor of things which *must* be true by definition as the latter doesn't require any further thinking on a future developer - also the latter might end up becoming untrue w/o people realizing.

Another example for me is `.get` on an Option, sure there are times where I *know* that it is `Some` and won't blow up but I'd much prefer to not do that for the reasons cited above. That said even I will drop my moral stance at times due to the pain in the ass factor :)